cryptographic_estimators/UOVEstimator/uov_problem.py

```diff
@@ -0,0 +1,87 @@
+# ****************************************************************************
+# Copyright 2023 Technology Innovation Institute
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# ****************************************************************************
+
+
+from ..base_problem import BaseProblem
+from sage.arith.misc import is_prime_power
+
```

Comment:
General comments:
1. Implement setters and getters for the `theta` (see MQProblem for reference)
2. Implement setters and getters for the `cost_hash` 
3. Implement getters for the number of variable number of polynomials and order of the field  (see MQProblem for reference)

Reply:
- Should `memory_bound` be an attribute of the UOVProblem class ?
- Should error handling for `theta` and `cost_hash` be handled only in the UOVProblem class, or should it also be implemented in the UOVAlgorithm class? I noticed that in MQ, error handling for `theta` is done in the MQAlgorithm class instead of the MQProblem class.

Reply:
`Should memory_bound be an attribute of the UOVProblem class ?` yes, but notice it is already in the base problem class.

`Should error handling for theta and cost_hash` only the problem class. The `MQEstimator` has to be updated in this regard. I'll do it.